src/widgetastic/widget/table.py

```diff
@@ -456,7 +456,10 @@ def __getitem__(self, item):
         if self.table_tree:
             nodes = self.resolver.glob(self.table_tree, '/table/tbody/tr*')
             at_index = at_index + 1 if self._is_header_in_body else at_index or 1
-            return next(n.obj for n in nodes if n.position == at_index)
+            try:
+                return six.next(n.obj for n in nodes if n.obj.index == at_index)
```

Comment:
This change to the obj index fixed my test behavior, but isn't appropriate for a rowspan/colspan table.

My test is related to dynamic table, and detecting rowspan/colspan in the first place. Going to reduce this to changing from `StopIteration` to `RowNotFound` here